### PR TITLE
Fix/late connected injection

### PR DESCRIPTION
**Pull request by quarkus**

### CODE REVIEW

**Comment by quarkus** [comment]
Hey @eddyloewen this is mainly tests / example to test / fix a certain edge case. 

We've had issues with "late provider" that would provide a "nullish" value during construction (before connection) which can happen as we are deferring the inital update by one frame. Also we've  had our "failover" function for late (defined) providers running to early. 

I basically split the functions and let the failover part run later. plus added some tests.

File: src/BaseElement.js [hunk 1]
```diff
@@ -497,6 +525,7 @@ class BaseElement extends HTMLElement {
                     // assign to prop
                     const eventPath = event.composedPath();
                     const target = eventPath[0] || event.target;
+                    // TODO do not assign when null
```
Comment (eddyloewen) [inline src/BaseElement.js]: @quarkus TODO?!

File: src/BaseElement.js [hunk 2]
```diff
@@ -87,7 +92,7 @@ class BaseElement extends HTMLElement {
      * This will define the attributes and properties as reactive getters and setter,
      * register observers, refs and events.
      */
-    connectedCallback() {
+    async connectedCallback() {
```
Comment (eddyloewen) [inline src/BaseElement.js]: @quarkus I know that we probably want this and also for other methods, but I don't know if it is a good idea to do it as a side effect in this PR.
Also I'm not entirely sure if this should be / would be / could be breaking?!

File: src/BaseElement.js (hunk 1 above)
Comment (quarkus) [reply src/BaseElement.js]: guess that was my first solution idea .. ill remove that

**Review by eddyloewen — APPROVED** [review]
